contracts/Core/StakeManager.sol

```diff
@@ -245,18 +241,17 @@ contract StakeManager is Initializable, ACL, StakeStorage, StateManager, Pause {
         require(locks[msg.sender][stakers[stakerId].tokenAddress].amount != 0, "Existing Lock doesnt exist");
 
         Structs.Staker storage staker = stakers[stakerId];
-        IStakedToken sToken = IStakedToken(stakers[stakerId].tokenAddress);
-
-        uint256 penalty = (staker.stake * parameters.resetLockPenalty()) / 100;
+        uint256 lockedAmount = locks[msg.sender][stakers[stakerId].tokenAddress].amount;
+        IStakedToken sToken = IStakedToken(staker.tokenAddress);
 
-        // Converting Penalty into sAmount
-        uint256 sAmount = _convertRZRtoSRZR(penalty, staker.stake, sToken.totalSupply());
+        uint256 penalty = (lockedAmount * parameters.resetLockPenalty()) / 100;
+        lockedAmount = lockedAmount - penalty;
```

Comment:
what if penalty > lockedAmount?

Reply:
for penalty > lockedAmount, resetLockPenalty will have to be greater 100 and since resetLockPenalty is in terms of percentage, penalty wont be greater than lockedAmount